cmd/cli/plugin/pinniped-auth/login_test.go

```diff
@@ -38,52 +38,129 @@ func Test_loginoidcCmd(t *testing.T) {
+				"--debug-session-cache", "true",
+				"--scopes", "openid",
+				"--ca-bundle", "/some/path",
+				"--ca-bundle-data", "somebase64encodeddata",
+				"--request-audience", "alternateaudience",
+				"--enable-concierge", "true",
+				"--concierge-namespace", "test-namespace",
+				"--concierge-authenticator-type", "webhook",
+				"--concierge-authenticator-name", "concierge-authenticator",
+				"--concierge-endpoint", "test-endpoint",
+				"--concierge-ca-bundle-data", "test-bundle",
+				"--concierge-api-group-suffix", "tuna.io",
+				"--concierge-is-cluster-scoped", "true",
+			},
+			wantArgs: []string{
+				"login",
+				"oidc",
+				"--issuer=different-issuer",
+				"--client-id=test-client",
+				"--listen-port=3737",
+				"--skip-browser=true",
+				"--debug-session-cache=true",
+				"--scopes=openid",
+				"--ca-bundle=/some/path",
+				"--ca-bundle-data=somebase64encodeddata",
+				"--request-audience=alternateaudience",
+				"--enable-concierge=true",
+				"--concierge-namespace=test-namespace",
+				"--concierge-authenticator-type=webhook",
+				"--concierge-authenticator-name=concierge-authenticator",
+				"--concierge-endpoint=test-endpoint",
+				"--concierge-ca-bundle-data=test-bundle",
+				"--concierge-api-group-suffix=tuna.io",
+				"--concierge-is-cluster-scoped=true",
+			},
 		},
 	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
+
+	for _, test := range tests {
+		test := test
+		// Resetting and setting the flags again is necessary in these tests to ensure new
+		// flags get picked up
+		loginCommand.ResetFlags()
+		setLoginCommandFlags()
+		t.Run(test.name, func(t *testing.T) {
 			var (
-				gotIssuer   string
-				gotClientID string
+				gotArgs []string
 			)
-			cmd := loginoidcCmd(func(execargs []string) error {
-				gotIssuer = execargs[2]
-				gotClientID = execargs[3]
+			cmd := loginOIDCCommand(func(args []string) error {
+				gotArgs = args
 
-				return tt.execReturnError
+				return test.execReturnError
 			})
 			require.NotNil(t, cmd)
 
 			var stdout, stderr bytes.Buffer
 			cmd.SetOut(&stdout)
 			cmd.SetErr(&stderr)
-			cmd.SetArgs(tt.args)
+			cmd.SetArgs(test.args)
 			err := cmd.Execute()
-			if tt.wantError {
+			if test.wantError {
 				require.Error(t, err)
 			} else {
 				require.NoError(t, err)
 			}
-			require.Equal(t, tt.wantStdout, stdout.String(), "unexpected stdout")
-			require.Equal(t, tt.wantStderr, stderr.String(), "unexpected stderr")
-			require.Equal(t, tt.wantIssuer, gotIssuer, "unexpected issuer")
-			require.Equal(t, tt.wantClientID, gotClientID, "unexpected client ID")
+			require.Equal(t, test.wantStdout, stdout.String(), "unexpected stdout")
+			require.Equal(t, test.wantStderr, stderr.String(), "unexpected stderr")
+			// using Subset method instead of EqualValues since not all args (like `sessionCachePath`) will be included in `wantArgs`
+			require.Subset(t, gotArgs, test.wantArgs, "Given CLI arguments not found in the actual CLI arguments")
```

Comment:
hmmm...shouldn't we just add `sessionCachePath` to `wantArgs`? i feel like a `require.Equal()` assertion here would give us the best test coverage.

Reply:
We can.  It would meaning adding a lot more to the `wantArgs` for other tests (all the default values), and the string argument order seems to matter for both `Equal` and `EqualValues`.

Reply:
Updated everything and changed it to `require.Equal`. Let me know what you think.  I think you're right, the value we get from making sure the arguments are exactly what we expect likely outweighs the pain of the order mattering.